src/engine/IndexScan.cpp

```diff
@@ -145,15 +149,15 @@ vector<ColumnIndex> IndexScan::resultSortedOn() const {
 std::optional<std::shared_ptr<QueryExecutionTree>>
 IndexScan::setPrefilterExprGetUpdatedQetPtr(
     std::vector<PrefilterVariablePair> prefilterVariablePairs) {
-  if (numVariables_ < 1) {
+  auto optSortedVarColIdxPair = getFirstSortedtVariableWithColumnIndex();
+  if (!optSortedVarColIdxPair.has_value()) {
     return std::nullopt;
   }
-
-  auto map = getVariableToSortedIndexMap();
+  const auto& [sortedVar, colIdx] = optSortedVarColIdxPair.value();
   for (auto& [prefilterExpr, variable] : prefilterVariablePairs) {
-    if (map.find(variable) != map.end()) {
+    if (sortedVar == variable) {
```

Comment:
more idiomatic here would be `std::ranges::find(prefilterVariablePairs, sortedVar, ad_utility::second)`
(And also less code).